submissions/NadiaVorontsova/a-tiny-JS-world-OOP/index.js

```diff
@@ -0,0 +1,96 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   
+   Code repository: https://github.com/NadiaVorontsova/a-tiny-JS-world-OOP
+   Web app: https://nadiavorontsova.github.io/a-tiny-JS-world-OOP/
+   */
+
+// ======== CLASS DEFINITIONS ========
+class Inhabitant {
+   constructor(species, name, gender, saying, friend) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.friend = friend || 'no friend:(';
+   }
+   printInhabitant() {
+      return [
+         "I am a <strong>" + this.species + "</strong>",
+         "My name is <strong>" + this.name + "</strong>",
+         "I am a <strong>" + this.gender + "</strong>",
+         "I say: <strong>" + this.saying + "</strong>",
+         "My friends: <strong>" + this.friend + "</strong>",
+      ].join("; ");
+   }
+}
+
+class Human extends Inhabitant {
+   constructor(name, gender, saying, friend) {
+      super('human', name, gender, saying, friend);
+      this.legs = 2;
+      this.hands = 2;
+   }
+   printHuman() {
```

Comment:
What if we could rename this method as `printInhabitant`? Same refers to `printAnimal`.
We probably would not need separate lists for people and animals at the end of the code.

There is a problem -- the method will try to call itself in an endless loop under current implementation due to **this**.printInhabitant.

See how [`super`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/super) can help you out.

Reply:
I did it) cool